src/code/z_scene.c

```diff
@@ -250,50 +264,52 @@ void Scene_CommandObjectList(PlayState* play, SceneCmd* cmd) {
     s32 i;
     s32 j;
     s32 k;
-    ObjectStatus* status;
-    ObjectStatus* status2;
-    ObjectStatus* firstStatus;
-    s16* objectEntry = SEGMENTED_TO_VIRTUAL(cmd->objectList.data);
+    ObjectEntry* entry;
+    ObjectEntry* invalidatedEntry;
+    ObjectEntry* entries;
+    s16* objectListEntry = SEGMENTED_TO_VIRTUAL(cmd->objectList.data);
     void* nextPtr;
 
     k = 0;
-    i = play->objectCtx.unk_09;
-    firstStatus = &play->objectCtx.status[0];
-    status = &play->objectCtx.status[i];
-
-    while (i < play->objectCtx.num) {
-        if (status->id != *objectEntry) {
-            status2 = &play->objectCtx.status[i];
-            for (j = i; j < play->objectCtx.num; j++) {
-                status2->id = OBJECT_INVALID;
-                status2++;
+    i = play->objectCtx.numPersistentEntries;
+    entries = play->objectCtx.slots;
+    entry = &play->objectCtx.slots[i];
+
+    while (i < play->objectCtx.numEntries) {
+        if (entry->id != *objectListEntry) {
+
+            invalidatedEntry = &play->objectCtx.slots[i];
+            for (j = i; j < play->objectCtx.numEntries; j++) {
+                invalidatedEntry->id = OBJECT_INVALID;
+                invalidatedEntry++;
             }
-            play->objectCtx.num = i;
+
+            play->objectCtx.numEntries = i;
             Actor_KillAllWithMissingObject(play, &play->actorCtx);
 
             continue;
         }
 
         i++;
         k++;
-        objectEntry++;
-        status++;
+        objectListEntry++;
+        entry++;
     }
 
-    ASSERT(cmd->objectList.length <= OBJECT_EXCHANGE_BANK_MAX,
+    ASSERT(cmd->objectList.length <= ARRAY_COUNT(play->objectCtx.slots),
```

Comment:
Noticed that the ASSERT here is flawed; It should be checking for `cmd->objectList.length + play->objectCtx.numPersistentEntries <= ARRAY_COUNT(play->objectCtx.slots)` instead

Reply:
good catch, though yeah we cant do anything about it cause its their mistake. can add a comment though

Reply:
@Dragorn421 can add this in this pr if you like, would even consider it an `@bug `